flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcReaders.java

```diff
@@ -127,6 +128,11 @@ public Integer nonNullRead(ColumnVector vector, int row) {
     @Override
     public DecimalData nonNullRead(ColumnVector vector, int row) {
       HiveDecimalWritable value = ((DecimalColumnVector) vector).vector[row];
+
+      // The hive ORC writer may will adjust the scale of decimal data.
+      Preconditions.checkArgument(value.precision() <= precision,
+          "Cannot read value as decimal(%s,%s), too large: %s", precision, scale, value);
```

Comment:
Does Flink require this?

Reply:
I think it is better to add this check to avoid potential precision mismatched bugs.